src/core/payment/PaymentSummarizer.ts

```diff
@@ -0,0 +1,69 @@
+import { injectable } from "inversify";
+import { Block } from "../dataProvider/dataprovider-types";
+import { PayoutTransaction } from "../payoutCalculator/Model";
+import { ISummarizer, PaymentProcess } from "./Model";
+
+@injectable()
+export class PaymentSummarizer implements ISummarizer<PaymentProcess> {
+    async calculateTotals(base: PaymentProcess) : Promise<void> {
+
+        const getTotalsFromBlocks = (blocks: Block[]) => {
+            let coinbasesum = 0
+            let feetransferfromcoinbasesum = 0
+            let usercommandtransactionfeessum = 0
+            blocks.forEach (block => {
+                coinbasesum += block.coinbase ?? 0
+                feetransferfromcoinbasesum += block.feetransferfromcoinbase ?? 0
+                usercommandtransactionfeessum += block.usercommandtransactionfees ?? 0
+            })
+
+            return {coinbasesum, feetransferfromcoinbasesum, usercommandtransactionfeessum}
+        }
+
+        const getTotalPayouts = (transactions: PayoutTransaction[]) => {
+            let amountsum = 0
+            let feesum = 0
+
+            transactions.forEach(transaction => {
+                amountsum += transaction.amount ?? 0
+                feesum += transaction.fee ?? 0
```

Comment:
This cannot actually be calculated until later. Initially, the send transaction fee is not set, because we don't know how many transactions will be sent until after we summarize them and handle substitutions and exclusions. So the send transaction fees will actually be established here:
https://github.com/jrwashburn/mina-pool-payout/blob/5db5fbb6f06ddd924c9d18b0d1f147aacbc8e7d5/src/core/transaction/TransactionBuilder.ts#L26
And this cannot be calculated until after that. 
It needs to be a two-part summarization because we want to summarize all payouts due (even if they won't be sent due to substitution or exclusion,) but cannot know the sending fees until they are excluded.